src/PoolManager.sol

```diff
@@ -294,6 +294,14 @@ contract PoolManager is IPoolManager, ProtocolFees, NoDelegateCall, ERC6909Claim
         _accountDelta(currency, paid.toInt128(), msg.sender);
     }
 
+    /// @inheritdoc IPoolManager
+    function clear(Currency currency, uint256 amount) external onlyWhenUnlocked {
+        int256 current = currency.getDelta(msg.sender);
+        if (current < 0) revert OwedFundsCannotBeCleared();
+        if (amount != uint256(current)) revert MustClearExactBalance();
```

Comment:
this basically only saves gas on ERC20 transfers then? because you still need to check how much is owed exactly to clear

Reply:
yeah but the difference between a tload (this check) and a transfer (which is 2 sstores) is huge